### PR TITLE
Move runner labels to `.env` and add unique identifiers

### LINKED ISSUE #19: Programmatically assigning labels to the runners from configuration files
The runners in this repo set a range of default runner labels and pass the REPOSITORY_URL and REGISTRATION_TOKEN variables to the config script.

It would be good to be able to override labels without having to re-build a Docker image! Consequently, it would be great to have one variable in the gh-actions-runner.env file which contains an environment variable with e.g. a comme-separated list of labels that's appended to the labels passed to the config script.

### PR DESCRIPTION
**Pull request by samuelburnham**
Closes #19

### CODE REVIEW

**Review by huitseeker — DISMISSED** [review]
🎉

File: scripts/entrypoint-wrapper.sh [hunk 1]
```diff
@@ -5,11 +5,23 @@
...
-RUNNER_DATE=$(date +"%y-%m-%d")
+RUNNER_DATE=$(date +"%m-%d-%y")
 head /dev/urandom | tr -dc a-f0-9 | head -c 13 | sed 's/$/-'$RUNNER_DATE'/' > /var/tmp/runner/_work/runner_id.txt
 _RUNNER_ID="$(cat /var/tmp/runner/_work/runner_id.txt)"
 RUNNER_NAME="${RUNNER_NAME}-${_RUNNER_ID}"
 RUNNER_WORKDIR="${RUNNER_WORKDIR}/${_RUNNER_ID}"
 
+# Get hardware info and append to labels
+LABELS="$(nproc --all) vCPUs"
+LABELS=${LABELS},"$(sudo dmidecode -t 19 | awk -F ': ' '/Range Size/{print $2}' | tail -n 1) RAM"
```
Comment (huitseeker) [inline scripts/entrypoint-wrapper.sh:16]: Is `dmidecode` installed?

File: scripts/entrypoint-wrapper.sh [hunk 2]
```diff
@@ -8,8 +8,23 @@ mkdir -p /var/tmp/runner/_work
 RUNNER_DATE=$(date +"%y-%m-%d")
 head /dev/urandom | tr -dc a-f0-9 | head -c 13 | sed 's/$/-'$RUNNER_DATE'/' > /var/tmp/runner/_work/runner_id.txt
 _RUNNER_ID="$(cat /var/tmp/runner/_work/runner_id.txt)"
+# The name is a bit long, but there's no other way to give the runner a unique name with a scaled docker-compose runner
 RUNNER_NAME="${RUNNER_NAME}-${_RUNNER_ID}"
 RUNNER_WORKDIR="${RUNNER_WORKDIR}/${_RUNNER_ID}"
 
+# Get hardware info and append to labels
```
Comment (samuelburnham) [inline scripts/entrypoint-wrapper.sh:15]: This creates some overlap with the downstream hardware env vars in Lurk, which ideally could be removed and accessed directly from `{{ env.LABELS }}` somehow

Comment (huitseeker) [reply scripts/entrypoint-wrapper.sh:15]: I think labels are fundamentally selectors for CI jobs.

While I like that this is much more x-platform than in prior iterations, those are strings that are not necessarily great as labels. 
- can `uname -ms` return something with a comma?
- can I really select against 121.57 GB RAM?
- is it wise to select against 32 vCPUs without also selecting on the CPU class?

IOW to me the labels would contain stuff that's much simpler (e.g. gpu label if there's one GPU, linux label if uname matches one of a few strings) than what we probe and display in CI.

File: scripts/entrypoint-wrapper.sh [hunk 3]
```diff
@@ -8,8 +8,19 @@ mkdir -p /var/tmp/runner/_work
...
 head /dev/urandom | tr -dc a-f0-9 | head -c 13 | sed 's/$/-'$RUNNER_DATE'/' > /var/tmp/runner/_work/runner_id.txt
 _RUNNER_ID="$(cat /var/tmp/runner/_work/runner_id.txt)"
+# The name is a bit long, but there's no other way to give the runner a unique name when scaled with docker-compose
 RUNNER_NAME="${RUNNER_NAME}-${_RUNNER_ID}"
 RUNNER_WORKDIR="${RUNNER_WORKDIR}/${_RUNNER_ID}"
 
+# Get basic hardware info and append to labels
+LABELS=${LABELS},$(uname -s),$(uname -m)
+if [ $(command -v nvidia-smi) ]; then
+        LABELS=${LABELS},"gpu",$(nvidia-smi --query-gpu=gpu_name --format=csv,noheader,nounits | tail -n 1)
```
Comment (huitseeker) [inline scripts/entrypoint-wrapper.sh:18]: 👍

**Review by huitseeker — APPROVED** [review]
🙏

**Pull request merged**